scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+
+        This function may return NaN when the spacing between floating point
+        values is too coarse to represent an argument for which the log-CCDF is
+        close to the provided `x`, even when the function is theoretically
+        continuous within the domain of the argument.
+
+        >>> X.ilogccdf(-np.inf), X.ilogccdf(-1e10), X.ilogccdf(-30.)
+        (0.5, nan, 0.4999999999998863)
+
+        """
+        return self._ilogccdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _ilogccdf_dispatch(self, x, method=None, **kwargs):
+        if self._overrides('_ilogccdf_formula'):
+            method = self._ilogccdf_formula
+        elif self._overrides('_ilogcdf_formula'):
+            method = self._ilogccdf_complementarity
+        else:
+            method = self._ilogccdf_inversion
+        return method
+
+    def _ilogccdf_formula(self, x, **kwargs):
+        return NotImplementedError(self._not_implemented)
+
+    def _ilogccdf_complementarity(self, x, **kwargs):
+        return self._ilogcdf_dispatch(_log1mexp(x), **kwargs)
+
+    def _ilogccdf_inversion(self, x, **kwargs):
+        return self._solve_bounded(self._logccdf_dispatch, x, kwargs=kwargs)
+
+    @_set_invalid_nan
+    def iccdf(self, x, *, method=None):
+        r"""Inverse complementary cumulative distribution function
+
+        The inverse complementary cumulative distribution function
+        :math:`G^{-1}(x)` is the argument :math:`y` for which the
+        complementary cumulative distribution function :math:`G(y)`
+        evaluates to :math:`x`.
+
+        .. math::
+
+            G^{-1}(x) = y \text{ such that } G(y) = x`
+
+        ``iccdf`` accepts `x` for :math:`x \in [0, 1]`.
+
+        Parameters
+        ----------
+        x : Array
+            The argument of the inverse complementary cumulative distribution
+            function (inverse CCDF).
+        method : {None, 'formula', 'complementarity', 'inversion'}
+            The strategy used to evaluate the inverse CCDF.
+            By default (``None``), the infrastructure chooses between the
+            following options.
+
+            - ``'formula'``: use a formula for the inverse CCDF itself
+            - ``'complementarity'``: evaluate the inverse CDF at the
+                                     complement of `x`; i.e. ``1 - x``.
+            - ``'inversion'``: solve numerically for the argument at which the
+                               CCDF is equal to `x`.
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The inverse CCDF evaluated at the provided argument.
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`. The
+        inverse CCDF takes on its minimum value of :math:`l` at :math:`x = 1`
```

Comment:
the maximum value at `l` for CCDF, no?

Reply:
The minimum value produced by inverse CDF is `l`. It does this when the *argument* is the maximum valid value, 1.0.

Reply:
I think throughout there is confusion about the phrase "takes on". It is the opposite of "takes in", not a typo. I will change it to "returns".

Reply:
You're right and I was wrong. The text if fine as is.

Reply:
Merging with https://github.com/mdhaber/scipy/pull/110#discussion_r1551401471